metals/src/main/scala/scala/meta/internal/metals/BloopServers.scala

```diff
@@ -402,29 +402,46 @@ final class BloopServers(
     }
   }
 
-  private def connectToLauncher(
-      bloopVersion: String,
-      bloopPort: Option[Int],
+  private def updateBloopJavaHomeBeforeLaunch(
       userConfiguration: UserConfiguration
-  ): Future[SocketConnection] = {
+  ) = {
+    def metalsJavaHome =
+      userConfiguration.javaHome
+        .orElse(sys.env.get("JAVA_HOME"))
+        .orElse(sys.props.get("java.home"))
     // we should set up Java before running Bloop in order to not restart it
     bloopJsonPath match {
       case Some(bloopPath) if !bloopPath.exists =>
         // we want to use the same java version as Metals, so it's ok to use java.home
-        val metalsJavaHome =
-          userConfiguration.javaHome
-            .orElse(sys.env.get("JAVA_HOME"))
-            .orElse(sys.props.get("java.home"))
-            // We can't cache java for nix/guix, since the path might change
-            .filterNot(path =>
-              path.startsWith("/nix/store/") || path.startsWith("/gnu/store/")
-            )
         writeJVMPropertiesToBloopGlobalJsonFile(
           userConfiguration.bloopJvmProperties.getOrElse(Nil),
           metalsJavaHome
         )
+      case Some(bloopPath) if bloopPath.exists =>
+        maybeLoadBloopGlobalJsonFile(bloopPath) match {
+          case (Some(javaHome), opts) =>
+            Try {
+              val homePath = AbsolutePath(Paths.get(javaHome))
+              // fix java home in case it changed
+              if (!homePath.exists)
+                writeJVMPropertiesToBloopGlobalJsonFile(
+                  opts,
+                  metalsJavaHome
+                )
```

Comment:
I would add here two `scribe.info`s here. First one just after if - `detected non existing java path in <path-to-bloop-config> file. The second one after overwriting it `overwrited it with <metalsJava>`

Reply:
Added!

Reply:
I also added info about which Java home and options is used currently for Bloop and where it is defined, so the users can potentially debug things easier themselves.

And also the first set up should be now logged.